### ISSUE TITLE
VS2015 complile error

### ISSUE THREAD

**Issue by CaffeMrDe** [niessner/mLib#11]
I try to complie the bundlefusion with vs2015, but it occur the problem about it . I thinks it compile problem, but how can I solve it? 

`
1>d:\3dre\bundlefusion\external\mlib\include\application-d3d11\d3d11canvas2d.h(302): error C2280: “ml::D3D11Canvas2D::ElementBillboard::ElementBillboard(const ml::D3D11Canvas2D::ElementBillboard &)”: 
`

Thanks you!

**Comment by dazinovic** [comment]
Are you using the current version of mlib? This should have been fixed in: include/application-d3d11/D3D11Canvas2D.h

**Comment by CaffeMrDe** [comment]
Yes. I get the source package with git clone the lastest master.

**Comment by dazinovic** [comment]
Can you write the exact compilation error that you get? Can you try to compile it with VS2019? I just tried it again with VS2019 and I am unable to reproduce the error that you get.

**Comment by CaffeMrDe** [comment]
I have already solve the problem . Thanks!

**Comment by tyronedong** [comment]
> 
> 
> I have already solve the problem . Thanks!

I met the same problem and tried a lot of method but didn't work. Could you please tell me how you solve it?

**Comment by chethanab16** [comment]
@CaffeMrDe could you tell me how to fix that error, I  am using visual studio 2017 I am getting the same error

**Comment by chethanab16** [comment]
Hi I try to compile the bundle fusion with visual studio 2017 but got the following error , could some one give me suggestion how to solve this problem.

![bundle_fusion_2017_error](https://user-images.githubusercontent.com/34049539/93157775-d2d1f280-f745-11ea-8bbe-d2ca9036905a.png)

**Comment by chethanab16** [comment]
**@dazinovic I got the Following error when i try to compile bundle fusion and I am using VS 2017 could you please suggest me  solution , thankyou.**

Severity	Code	Description	Project	File	Line	Suppression State
Error	C2280	'Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false>::operator =(const Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\mlibexternal\include\eigen\src\core\solvetriangular.h	189	
Error	C2280	'Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false>::operator =(const Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\mlibexternal\include\eigen\src\core\solvetriangular.h	189	
Error	C2280	'ml::MeshData<float> &ml::MeshData<float>::operator =(const ml::MeshData<float> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\bundlefusion-master\external\mlib\include\core-mesh\meshdata.cpp	575	
Error	C2280	'Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false>::operator =(const Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\mlibexternal\include\eigen\src\core\solvetriangular.h	189	
Error	C2280	'Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false>::operator =(const Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\mlibexternal\include\eigen\src\core\solvetriangular.h	189	
Error	C2280	'Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false>::operator =(const Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\mlibexternal\include\eigen\src\core\solvetriangular.h	189	
Error	C2280	'Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false>::operator =(const Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\mlibexternal\include\eigen\src\core\solvetriangular.h	189	
Error	C2280	'ml::MeshData<float> &ml::MeshData<float>::operator =(const ml::MeshData<float> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\bundlefusion-master\external\mlib\include\core-mesh\meshdata.cpp	575	
Error	C2280	'ml::MeshData<float> &ml::MeshData<float>::operator =(const ml::MeshData<float> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\bundlefusion-master\external\mlib\include\core-mesh\meshdata.cpp	575	
Error	C2280	'Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false>::operator =(const Eigen::Block<Eigen::Block<Eigen::Map<Derived,0,Eigen::Stride<0,0>>,-1,-1,false>,-1,-1,false> &)': attempting to reference a deleted function	FriedLiver	d:\azur_kinect_bundlefusion_2017_port\mlibexternal\include\eigen\src\core\solvetriangular.h	189

**Comment by CaffeMrDe** [comment]
I guess the problem what cause is the VS complier version. I change the complier version to vs2013, and it solve.

**Comment by chethanab16** [comment]
@CaffeMrDe  thank you for your suggestion, I  want to run the project  on VS2017 or Vs 2015 Compiler , could you give me some suggestion on that , thankyou
